### ISSUE TITLE
Bottom of screen gets cut off in full screen mode - Windows 10, 1366x768

### ISSUE THREAD

**Issue by NixieLake** [AppleWin/AppleWin#464]
I've noticed when I enter full screen mode, the bottom line of text is partially cropped at the bottom. I've been looking around to see if there was any fix for it but couldn't find anyone else complaining about the same problem.

I'm running it on a Windows 10 laptop, Acer Aspire F 15 using Intel HD graphics 620 (onboard).

Thanks

**Comment by NixieLake** [comment]
Clarification: After comparing each model, I could confirm that it's consistent across each and that the amount of display that seems to be cropped is about 1 or 2 rows of pixels. I'm thinking the full screen mapping might be shifted 1 or 2 rows down because there's a narrow blank space above the emulated display.

**Comment by sicklittlemonkey** [comment]
What version of AppleWin are you using?

Cheers,
Nick.

**Comment by NixieLake** [comment]
AppleWin ver. 1.26.1.1

**Comment by Michaelangel007** [comment]
I'll try setting either my Win7 box or my VMware box to 1366x768 to see if I can repo. but in the meantime is it possible to get a screenshot of this?  Thx.

**Comment by tomcw** [comment]
FYI, using 1.26.3.0, my VAIO laptop at 1600x900 is fine.
I then set the resolution to 1366x768, and yes, the very bottom pixel line is missing.

**Comment by tomcw** [comment]
Full screen:
(In Paint, I zoomed in at the top, and there is a 5 pixel vertical gap between the edge and the start of the 1st char. Each Apple pixel is 4 Windows pixels, so it seem that the rendering is starting 1 Apple pixel lower than it should.)  
![text](https://user-images.githubusercontent.com/6696896/29477719-7ee28626-8461-11e7-8e28-5b7efbf771c7.png)

Windowed:
![text-windowed2](https://user-images.githubusercontent.com/6696896/29477840-1712ae1c-8462-11e7-8b7e-84579bc27d3e.png)

**Comment by Michaelangel007** [comment]
Sorry, been knee-deep in debugger fixes / enhancements.  Will try to remember to take a look "soonish".

At least I have that bloody retina resolution for VMWare turned off (aka, pixel doubling is on) so I don't have to squint trying to see a itsy, bitsy, tiny window now.

**Comment by Michaelangel007** [comment]
Confirmed on Windows 7, 1366x768, AppleWin 1.26.3.0

* Extra 5 px of black on top
* Bottom 5 px on bottom are cut off

![win7_1366x768](https://user-images.githubusercontent.com/7876796/29745673-6e7cd33e-8a76-11e7-98df-8c8df62a246e.png)

**Comment by Michaelangel007** [comment]
Ninja'd !  Bug in `SetFullScreenMode()`, [Line 2181])(https://github.com/AppleWin/AppleWin/blob/master/source/Frame.cpp#L2181)

```
	viewporty  = VIEWPORTY;
```

Fix is trivial 1-line patch:

```
	viewporty  = g_bIsFullScreen ? 0 : VIEWPORTY;
```

**Comment by Michaelangel007** [comment]
Vars:

```
A2_WINDOW_HEIGHT: 384
A2_WINDOW_WIDTH : 560

width: 1366
height: 768

scalex: 2
scaley: 2

top : 0
left: 0
```

**Comment by Michaelangel007** [comment]
Tagging as 1.27 as a reminder that `bin/History.txt` needs to be updated.

**Comment by Michaelangel007** [comment]
Is this going to break the fullscreen crosshairs ?

**Comment by tomcw** [comment]
>Is this going to break the fullscreen crosshairs (Joystick) ?

I now get an assert for the Mousecard with "Restrict mouse to Apple window" **unchecked**.
I get the assert here:
```
static void FrameSetCursorPosByMousePos(int x, int y, int dx, int dy, bool bLeavingAppleScreen)
{
   :
	else	// Mouse entering Apple screen area
	{
		:
		_ASSERT(y <= g_nViewportCY);
```

I'll take a deeper look.

**Comment by tomcw** [comment]
>Is this going to break the fullscreen crosshairs (Joystick) ?  DrawCrosshairs ()  assumes viewport is >= 5 ...

I checked `void DrawCrosshairs (int x, int y)`, when the `GetFullScreenOffsetY()` returns 0 - ie. the full-screen height is 768 (using -fs-height=768).

Anyway this causes `OffsetRect()` to set `rect.top = -5` (return value TRUE / succeeded) and `FillRect()` to return 1 (succeeded).

So the crosshairs at the top and bottom just don't get draw, but otherwise there's no ill effect.

I still need to look at that mouse _ASSERT I reported above.

NB. This only happens when `GetFullScreenOffsetY()` returns 0, which is only true for a display height of 384 or 768.
